### ISSUE TITLE
image galaxy

### ISSUE THREAD

**Issue by bbernhard** [bbernhard/imagemonkey-core#63]
One thing that's on my mind for quite a long time is some sort of "image galaxy". Like as you are flying through space and exploring the dataset during that flight. (could also be a cool idea for a VR app)

Here is a small demo I created over the weekend: 

![image_galaxy](https://user-images.githubusercontent.com/5710631/35587795-1a8a01f2-05ff-11e8-99c6-c3faf79b87b1.gif)

![image_galaxy2](https://user-images.githubusercontent.com/5710631/35587997-a83be29a-05ff-11e8-82a8-0dea6bc62c00.gif)

I am not happy yet with the user experience, but yeah ;-) It's lagging quite a lot in the GIF...it's not that bad in reality, but still not as smooth as I wish it would be. The GIF is also x2 in terms of speed ;)

**Comment by dobkeratops** [comment]
absolutely spot on! i'm interested in this too. SO much you could do with it, especially with links (imagine tendrils connecting related ones - or nearing one animating the whole thing toward a different priority (switching dimensions) .. or the images themselves being portals to spaces where the spatial sort emphasises them
i imagine colour coding on the links, the images perhaps enclosed in bubbles, etc..

**Comment by dobkeratops** [comment]
... and this is why I thought about the *thumbnail serving* - your 'image galaxy' browser would get the thumbnails first, and read in the hi-res version of the examples you fly toward

**Comment by dobkeratops** [comment]
> I am not happy yet with the user experience, but yeah ;-)

I'm sure even web-gl/JS would handle this fine once you arrange it right

**Comment by dobkeratops** [comment]
>> (could also be a cool idea for a VR app)

yes, absolutely

I can tell you another use case I have in mind for myself: bike-trainer (which could of course be VR or just on a screen) - motion tied to pedalling, but track your gaze (which VR would do best) to shift where you travel to - allowing exploring of a large dataset while you're sat pedalling on the indoor trainer. 
I'd also thought about trying that as an explorer for linked text, but i wasn't sure you'd be able to read so much with 3d motion going on.

**Comment by dobkeratops** [comment]
other ideas 'image forest', 'image city' - but the 'image galaxy' utilizes space better.

**Comment by bbernhard** [comment]
You are right, there are endless possibilities. ;) My main motivation was to give the famous three.js a try (always wanted to play with it a bit). Unfortunately I quickly realized that it's pretty far away from the technology (concepts) I am used to...the whole scene/geometry/camera thing is still giving me headaches :D 

So I am really hoping that someone else will pick up the idea first, so that I finally can get myself VR glasses to browse the image gallery that way ;) 

> ... and this is why I thought about the thumbnail serving - your 'image galaxy' browser would get the thumbnails first, and read in the hi-res version of the examples you fly toward

I did something similar for the test, I used the APIs `width` and `height` query parameter to scale the images down to ~ 100x100px. Although there is no caching in place, it performed pretty decent...to be honest I was expecting the performance to be far worse. So I think with some caching and further improvements it should perform even better.

**Comment by dobkeratops** [comment]
>> So I am really hoping that someone else will pick up the idea first, so that I finally can get myself VR glasses to browse the image gallery that way

I wonder if there are any general purpose 'VR graph browsers' (linked images+text).. if not I think we both agree on what it could do

I haven't personally jumped on VR as yet, because I prefer to spend time in the desktop or real world. still, the use case I have in mind ('something to keep you interested whilst on an indoor bike-trainer') would definitely benefit from head tracking. 

I'm looking forward to the day when VR resolution is high enough that a VR virtual desktop setup can really replace desktop monitors ..

**Comment by bbernhard** [comment]
> I wonder if there are any general purpose 'VR graph browsers' (linked images+text).. if not I think we both agree on what it could do

Totally agreed. I already look a bit on the internet, but couldn't find anything useful. But I think I am still using the wrong search terms, as I can't imagine that there aren't such projects out there. 

> I haven't personally jumped on VR as yet, because I prefer to spend time in the desktop or real world. still, the use case I have in mind ('something to keep you interested whilst on an indoor bike-trainer') would definitely benefit from head tracking.

> I'm looking forward to the day when VR resolution is high enough that a VR virtual desktop setup can really replace desktop monitors for reading text .. at that point I would want to use VR all the time

Yeah, me too!
